compiler/pom.xml

```diff
@@ -139,6 +139,18 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
```

Comment:
I believe this adds no runtime dependencies. Or do you mean excluding libraries from being shaded into the jar?